lib/finch/request.ex

```diff
@@ -49,18 +60,28 @@ defmodule Finch.Request do
   """
   @type body() :: iodata() | {:stream, Enumerable.t()} | nil
 
+  @typedoc """
+  Optional metadata of the request. This field can be used by libraries that build on top of Finch
+  to inject information about the request that needs to be retrieved later on, for example, from
+  handlers that consume `Finch.Telemetry` events.
+  """
+  @type metadata() :: %{optional(atom()) => term()}
+
   @type t :: %__MODULE__{
-          scheme: Mint.Types.scheme(),
+          body: body(),
```

Comment:
I now see that I changed the ordering here accidentally :) (it's because the docs display it in an alphabetical order, not sure if you want to keep it this way)